File: leet/0062-unique-paths/unique-paths.go
func uniquePaths(m int, n int) int {
    cache := make([][]int, m)
    for i := 0; i < m; i++ {
        cache[i] = make([]int, n)
    }
    
    for i := 0; i < m; i++{
        for j:= 0; j < n; j++{
            if i == 0 || j == 0 {
                cache[i][j] = 1
            } else {
                cache[i][j] = cache[i - 1][j] + cache[i][j - 1]
            }
        }
    }
    
    return cache[m -1][n - 1]
}

File: leet/0062-unique-paths/unique-paths.js
/**
 * @param {number} m
 * @param {number} n
 * @return {number}
 */
var uniquePaths = function(m, n) {
    let mp = new Map();

    let dfs = function(i,j){
        if (i == 1 || j == 1) return 1

        let key = i + "-" + j
        if (mp.has(key)) return mp.get(key)

        let ret = dfs(i - 1, j) + dfs(i, j -1)
        mp.set(key, ret)
        return ret
    }

    return dfs(m,n)
};
